packages/demo-protobuf/src/demo.spec.ts

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import { assert } from "@cosmjs/utils";
+import protobuf from "protobufjs";
+
+import reflectionRoot from "./demo";
+import demoJson from "./demo.json";
+import demoProto from "./demo.proto";
+import { cosmos_sdk as cosmosSdk, google } from "./generated/codecimpl";
+
+type MsgDemo = {
+  readonly example: string;
+};
+
+const { Coin } = cosmosSdk.v1;
+const { TxBody } = cosmosSdk.tx.v1;
+const { MsgSend } = cosmosSdk.x.bank.v1;
+const { Any } = google.protobuf;
+
+function getTypeName(typeUrl: string): string {
+  const parts = typeUrl.split(".");
+  return parts[parts.length - 1];
+}
+
+describe("protobuf demo", () => {
+  it("works with generated static code", () => {
+    const coin = Coin.create({
+      denom: "ucosm",
+      amount: "1234567890",
+    });
+    const msgSend = MsgSend.create({
+      fromAddress: Uint8Array.from(Array.from({ length: 20 }, () => 1)),
+      toAddress: Uint8Array.from(Array.from({ length: 20 }, () => 2)),
+      amount: [coin],
+    });
+    const msgSendBytes = MsgSend.encode(msgSend).finish();
+    const msgSendWrapped = Any.create({
+      type_url: "/cosmos.bank.MsgSend",
+      value: msgSendBytes,
+    });
+    const txBody = TxBody.create({
+      messages: [msgSendWrapped],
+      memo: "Some memo",
+      timeoutHeight: 9999,
+      extensionOptions: [],
+    });
+    const txBodyBytes = TxBody.encode(txBody).finish();
+
+    // Deserialization
+    const txBodyDecoded = TxBody.decode(txBodyBytes);
+    const msg = txBodyDecoded.messages[0];
+    assert(msg.value);
+    const msgSendDecoded = MsgSend.decode(msg.value);
+
+    // fromAddress and toAddress are now Buffers
+    expect(Uint8Array.from(msgSendDecoded.fromAddress)).toEqual(msgSend.fromAddress);
+    expect(Uint8Array.from(msgSendDecoded.toAddress)).toEqual(msgSend.toAddress);
+    expect(msgSendDecoded.amount).toEqual(msgSend.amount);
+  });
+
+  it("works with dynamically loaded proto files", () => {
+    const { root } = protobuf.parse(demoProto);
+    const typeUrl = "/demo.MsgDemo";
+    const encoder = root.lookupType(getTypeName(typeUrl));
```

Comment:
Oh, this works without the fully qualified name (just `MsgDemo` from `demo.MsgDemo`)

Reply:
Yes, I think because the root is attached to the namespace `demo`.